src/app/services/todo.service.ts

```diff
@@ -1,19 +1,41 @@
 import { ITodo } from 'src/app/models/ITodo';
 import { Injectable } from '@angular/core';
 import { HttpService } from './http.service';
-import { Observable } from 'rxjs';
-import { map } from 'rxjs/operators';
+import { Observable, BehaviorSubject, of } from 'rxjs';
+import { map, catchError } from 'rxjs/operators';
 import { RandomService } from './random.service';
 
 @Injectable({
   providedIn: 'root',
 })
 export class TodoService {
-  public todoList: ITodo[];
+  public todoList: ITodo[] = [];
+  private subject = new BehaviorSubject<ITodo[]>([]);
+  public todos$ = this.subject.asObservable();
   private url = 'https://jsonplaceholder.typicode.com/todos';
 
   constructor(private httpService: HttpService, private randomService: RandomService) {}
 
+  public getTodos(): Observable<ITodo[]> {
+    return this.httpService.get<ITodo[]>(this.url).pipe(
+      map((items) => {
+        const fullItems = items.map((item) => {
+          item.creationDate = this.randomService.getRandomDate(2020, [1, 4]);
+          item.deadline = this.randomService.getRandomDate(2020, [5, 11]);
+          item.username = this.randomService.getRandomName();
+          return item;
+        });
+        this.todoList = fullItems;
+        this.subject.next(fullItems);
```

Comment:
move this.todoList = fullItems;
and this.subject.next(fullItem) to tap method, after map